### PR TITLE
EIP-155 implementation

### PR DESCRIPTION
**Pull request by splix**
Based on ethereum/go-ethereum@dc2e34d

Just applied and unit-tested EF implementation, need to double check the code now, and also going to run in on testnet together with parity. 

Submitted just for review only, please do not merge it until it would be fully tested

### CODE REVIEW

**Comment by elaineo** [comment]
@realcodywburns are you referring to core/config.go?

**Comment by realcodywburns** [comment]
In the params/protocol_params.go. For eth the testnet chains and main chains are diferentiated with chain ids, main net is 1, mordern is 2, romp is 3. It is to prevent replay attacks, however unlikely, between even those nets.

File: core/types/transaction.go [hunk 1]
```diff
@@ -49,24 +48,28 @@ type txdata struct {
 	Recipient       *common.Address `rlp:"nil"` // nil means contract creation
 	Amount          *big.Int
 	Payload         []byte
-	V               byte     // signature
-	R, S            *big.Int // signature
+	V, R, S         *big.Int // signature
```
Comment (elaineo) [inline core/types/transaction.go]: what was the rationale behind changing V from `byte` to `big.Int`? sorry if i missed this in the Slack somewhere

File: core/types/transaction_signing.go [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+		tx.data.Price,
+		tx.data.GasLimit,
+		tx.data.Recipient,
+		tx.data.Amount,
+		tx.data.Payload,
+		s.chainId, uint(0), uint(0),
+	})
+}
+
+func (s ChainIdSigner) SigECDSA(tx *Transaction, prv *ecdsa.PrivateKey) (*Transaction, error) {
```
Comment (elaineo) [inline core/types/transaction_signing.go]: is this function supposed to be here?

File: core/types/transaction.go (hunk 1 above)
Comment (splix) [inline core/types/transaction.go]: I guess because it's used in operations with other bit.Ints, so should be converted at some point before that

**Comment by splix** [comment]
@realcodywburns for testnet we can use 4. is that ok?

File: core/types/transaction_signing.go (hunk 2 above)
Comment (splix) [inline core/types/transaction_signing.go]: I see it's used from other places. Why do you think it shouldn't be here?

**Comment by realcodywburns** [comment]
I was thinking 62. Its close enough to ours that if a  type mismatch. Like byte in the Java client being unable to accept over 127

**Comment by splix** [comment]
@realcodywburns hm, that's a good reason to have 62, yeah. My reason for 4 was that all testnets would be in same range

File: core/types/transaction_signing.go (hunk 2 above)
Comment (splix) [inline core/types/transaction_signing.go]: Ah, you mean that it's same just call another function, so looks different from BasicSign? Yeah, it seems that it requires some refactoring

Comment (elaineo) [inline core/types/transaction_signing.go]: yeah, it's not a big deal, just seems like an unnecessary piece

**Comment by splix** [comment]
@realcodywburns @elaineo please see updated code

**Comment by elaineo** [comment]
@splix changes look good to me. I think the ` SigECDSA` function might be an accident that should not exist

**Comment by splix** [comment]
Running on testnet starting from block 1915000 (expected tomorrow)

**Pull request merged**